autometa/common/external/hmmer.py

```diff
@@ -234,10 +230,30 @@ def translater(x): return translations.get(x, x.rsplit('_', 1)[0])
     return outfpath
 
 
-def main(args):
+def main():
+    import argparse
+    import logging as logger
+    logger.basicConfig(
+        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
+        datefmt='%m/%d/%Y %I:%M:%S %p')
+    parser = argparse.ArgumentParser(
+        description='Retrieves markers with provided input assembly')
+    parser.add_argument('orfs', help='</path/to/assembly.orfs.faa>')
+    parser.add_argument('hmmdb', help='</path/to/hmmpressed/hmmdb>')
+    parser.add_argument('cutoffs', help='</path/to/hmm/cutoffs.tsv>')
+    parser.add_argument('hmmscan', help='</path/to/hmmscan.out>')
+    parser.add_argument('markers', help='</path/to/markers.tsv>')
+    parser.add_argument('--log', help='</path/to/parallel.log>')
+    parser.add_argument('--force', help="force overwrite of out filepath",
+                        action='store_true')
+    parser.add_argument('--cpus', help='num cpus to use', default=0, type=int)
+    parser.add_argument(
+        '--parallel', help="Enable GNU parallel", action='store_true')
+    parser.add_argument('--verbose', help="add verbosity", action='store_true')
+    args = parser.parse_args()
+
     if args.verbose:
         logger.setLevel(logger.DEBUG)
-
     try:
         result = hmmscan(
             orfs=args.orfs,
```

Comment:
>2. In the main function, we run the hmmscan function in a try/except block. However because the force command line argument is passed to the function, I'm not sure if FileExistsError will ever be encountered here. We should perhaps handle any other exception here.

Has this been addressed?

Reply:
I actually commented about in the PR above. Copying from above,
_"Did not remove FileExistsError from the except block of main as that will be needed if --force is not passed. I could not identify any other error to catch/raise."_

Reply:
What I meant was that the `hmmscan` function has error handling for `FileExistsError`, so this error will never be encountered by `main`.

Reply:
So, no @Sidduppal it is still not addressed.

Reply:
I am not sure about that, because `hmmscan` is just rasing the error not handling it.

Reply:
The `hmmscan` function raises a `FileExistsError` when the output exists and the user has not specified `force = True`. That means the program will exit. I don't know what you mean by handling it? The user has the choice to specify `force` or not. If you just go ahead and overwrite that is choosing for them (you may as well not have that argument), and the only alternative is stopping the program and informing the user why. My complaint is that as written, since this error is thrown by `hmmer`, then `main` will never encounter a `FileExistsError`, so that try/except block should be taken out. Leaving it in is unnecessarily confusing to developers. Say for instance someone in the future wants to alter the error handling here when running through command line. If they change `main` they will be confused why the behavior doesn't change. The answer is they were looking in the wrong place!